src/ReportPortal.Shared/Reporter/LaunchReporter.cs

```diff
@@ -105,7 +105,7 @@ public void Start(StartLaunchRequest request)
                 throw exp;
             }
 
-            if (_rerunOfUuid != null)
+            if (_rerunOfUuid != null && _isRerun)
```

Comment:
If propose restructure conditions.

```csharp
if (rerun)
{
  request.IsRerun = true;

  if (_rerunOfUuid != null) request.RerunOfLaunchUuid = _rerunOfUuid;

  ...
}
```

Reply:
Good point! We can actually remove duplication and simplify condition. Please have a look